inst/templates/aggregate_format_hydro_data.R

```diff
@@ -0,0 +1,129 @@
+# Meta data ----
+# A template analysis file from erieacoustics
+# Version 0.0.1
+# September 27, 2022
+# M. DuFour
+
+# What it does ----
+# This script will import, combine, and format summarized integration and target strength
+# data from hydroacoustic survey transects. Three .csv files with data associated by
+# BASIN, STRATUM, GRID, LAYER (epilimnion vs. hypolimnion region), and interval are exported.
+#
+# 'hacdat.csv' includes  integration, mean target strength, and fish density estimates.
+#
+# 'histo.csv' includes single target counts for 0.5 dB target strength bins ranging
+# from -64.5 to -20.5 dB.
+#
+# histohac.csv' merges 'hacdat' and 'histo' together.
+
+# How to use ----
+# 1. Ensure that data has imported to Echoview template using '1_import_data_to_template.R'.
+# 2. Ensure that Echoview files have been processed and scrutinized
+# 3. Ensure that transect level data has been exported using '2_export_data_from_EV'
+# 4. Run script below
+
+# ----
+
+## double check that all required packages are installed
+pck_list <- c('dplyr','readr','magrittr','base','magrittr','stats','tidyr')
+
+is_installed <- pck_list %in% installed.packages()
+if(!all(is_installed)){
+  missing <- pck_list[!is_installed]
+  stop(paste0("\nuse install.packages(", missing,") to install ", missing," package"))
+}
+
+
+## load packages
+library(readr)
+library(dplyr)
+library(base)
+library(magrittr)
+library(stats)
+library(tidyr)
+
+
+## Gather a list of all files from /3_Ping_Data directory
+allfiles<-base::dir('3_Ping_Data', recursive = T, full.names = T)
+
+## Import and bind together integration data
+intg<-base::grep(allfiles, pattern = "intg\\.csv$", value = T)
+intg<-dplyr::bind_rows(base::lapply(intg, readr::read_csv))
+intg<-intg %>% dplyr::filter(!(base::is.na(Region_name)))
+
+## Import and bind together target strength data
+ts<-base::grep(allfiles, pattern = "ts\\.csv$", value = T)
+ts<-dplyr::bind_rows(base::lapply(ts, readr::read_csv))
+
+## Merge/join integration and target strength data
+hacdat <- dplyr::left_join(intg, ts, by=c('Region_name', 'Interval', 'Date_M', 'Time_M'))
+
+## remove intervals less than 375 m - typically shorts segment collected beyond specified 5 km transect
+hacdat$IntDist <- hacdat$Dist_E.x-hacdat$Dist_S.x
```

Comment:
Good call. The *.x field names are annoying!

Reply:
I went with the first suggestion and making edits to the aggregate_format_... scripts.

Reply:
if they are included in the by=c() statement it might be a good idea to add a check that makes sure all the fields were joined. I think EV uses the same ping bounds for each export and thus the values in those field should be an identical match  but it's worth having some form of check in place in case there is some variation between the output files. The join won't work unless it is an identical match.
``` r
foo <- data.frame(A = c("A", "B"), B = c(0.333333333, 0.6))
bar <- data.frame(C = c("A", "B"), B = c(0.3333333333, 0.6))
dplyr::left_join(foo, bar)
#> Joining, by = "B"
#>   A         B    C
#> 1 A 0.3333333 <NA>
#> 2 B 0.6000000    B
```